src/m365/entra/commands/group/group-member-remove.ts

```diff
@@ -0,0 +1,260 @@
+        userNames: typeof args.options.userNames !== 'undefined',
+        subgroupIds: typeof args.options.subgroupIds !== 'undefined',
+        subgroupNames: typeof args.options.subgroupNames !== 'undefined',
+        role: typeof args.options.role !== 'undefined',
+        suppressNotFound: !!args.options.suppressNotFound,
+        force: !!args.options.force
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-i, --groupId [groupId]'
+      },
+      {
+        option: '-n, --groupName [groupName]'
+      },
+      {
+        option: '--userIds [userIds]'
+      },
+      {
+        option: '--userNames [userNames]'
+      },
+      {
+        option: '--subgroupIds [subgroupIds]'
+      },
+      {
+        option: '--subgroupNames [subgroupNames]'
+      },
+      {
+        option: '-r, --role [role]',
+        autocomplete: this.roleValues
+      },
+      {
+        option: '--suppressNotFound'
+      },
+      {
+        option: '-f, --force'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.groupId !== undefined && !validation.isValidGuid(args.options.groupId)) {
+          return `'${args.options.groupId}' is not a valid GUID for option 'groupId'.`;
+        }
+
+        if (args.options.userIds !== undefined) {
+          const invalidGuids = validation.isValidGuidArray(args.options.userIds);
+          if (invalidGuids !== true) {
+            return `Invalid GUIDs found for option 'ids': ${invalidGuids}.`;
+          }
+        }
+
+        if (args.options.userNames !== undefined) {
+          const invalidUpns = validation.isValidUserPrincipalNameArray(args.options.userNames);
+          if (invalidUpns !== true) {
+            return `Invalid UPNs found for option 'userNames': ${invalidUpns}.`;
+          }
+        }
+
+        if (args.options.subgroupIds !== undefined) {
+          const invalidGuids = validation.isValidGuidArray(args.options.subgroupIds);
+          if (invalidGuids !== true) {
+            return `Invalid GUIDs found for option 'subgroupIds': ${invalidGuids}.`;
+          }
+        }
+
+        if (args.options.role !== undefined && this.roleValues.indexOf(args.options.role) === -1) {
+          return `Option 'role' must be one of the following values: ${this.roleValues.join(', ')}.`;
+        }
+
+        if ((args.options.subgroupIds !== undefined || args.options.subgroupNames !== undefined) && args.options.role?.toLowerCase() !== 'member') {
+          return `When removing subgroups, the 'role' option must be set to 'Member'.`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push(
+      { options: ['groupId', 'groupName'] },
+      { options: ['userIds', 'userNames', 'subgroupIds', 'subgroupNames'] }
+    );
+  }
+
+  #initTypes(): void {
```

Comment:
Do we really need to specify all these types explicitly here?

Reply:
With Minimist, I usually type every option to the type we expect it to be. If we don't do this, and I pass `25` as `ids`. The validation will fail because we treat it as a string while Minimist parses it into a number.

If you want, I can try to convert everything to Zod.

Reply:
Understood. Let's leave it for now and merge as-is, and take zod refactoring separately to avoid further delay.